### ISSUE TITLE
NullReferenceException when calling SetNode()

### ISSUE THREAD

**Issue by mgilski** [YarnSpinnerTool/YarnSpinner#251]
**What is the current behavior?**

Yarn throws `NullReferenceException` when trying to call `SetNode()`.

**Please provide the steps to reproduce, and if possible a minimal demo of the problem**:

Run this code:
```
public static void Main()
        {
            Compiler.CompileFile("Test.yarn",
                             out Program program,
                             out IDictionary<string, StringInfo> stringTable);
            var dialogue = new Dialogue(new DialogueVariables());
            dialogue.SetProgram(program);
            dialogue.SetNode("Start");
        }
```
where `DialogueVariables` implements `VariableStorage` and `Test.yarn` is the one from the repo.

**Please tell us about your environment:**
  
  - Yarn Spinner Version: 1.2.0
  - System: Ubuntu 16.04
  - Framework: Mono 6.8.0.123

**Other information** 

My guess is this has something to do with me using mono, because this can't be normal. Unfortunately I can't test it with .net, because I'm running Linux on a Chromebook and I'm reeealy low on disk space.

Stack trace:
```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object.
  at Yarn.VirtualMachine.SetNode (System.String nodeName) [0x00087] in <4e437c68f0094b76b6acc31d81816ac9>:0 
  at Yarn.Dialogue.SetNode (System.String startNode) [0x00001] in <4e437c68f0094b76b6acc31d81816ac9>:0 
  at DialogRenderer.OnDialogueRenderingStarted (System.String dialoguePath) [0x00047] in /home/maciej/station/Station/src/gui/DialogRenderer.cs:39 
  at (wrapper managed-to-native) Godot.NativeCalls.godot_icall_2_585(intptr,intptr,string,object[])
  at Godot.Object.EmitSignal (System.String signal, System.Object[] args) [0x00001] in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Object.cs:354 
  at ExtensionMethods.EmitSignal[T] (Godot.Object node, System.Object[] args) [0x00000] in /home/maciej/station/Station/src/ExtensionMethods.cs:27 
  at Npc.OnDialogueStarted () [0x0001d] in /home/maciej/station/Station/src/actors/npc/Npc.cs:32 
  at (wrapper managed-to-native) Godot.NativeCalls.godot_icall_2_585(intptr,intptr,string,object[])
  at Godot.Object.EmitSignal (System.String signal, System.Object[] args) [0x00001] in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Object.cs:354 
  at ExtensionMethods.EmitSignal[T] (Godot.Object node, System.Object[] args) [0x00000] in /home/maciej/station/Station/src/ExtensionMethods.cs:27 
  at NpcCollider.Interact () [0x00001] in /home/maciej/station/Station/src/actors/npc/NpcCollider.cs:11 
  at Interactable.OnControllerInteract () [0x0000b] in /home/maciej/station/Station/src/objects/Interactable.cs:21 
  at (wrapper managed-to-native) Godot.NativeCalls.godot_icall_2_585(intptr,intptr,string,object[])
  at Godot.Object.EmitSignal (System.String signal, System.Object[] args) [0x00001] in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Object.cs:354 
  at ExtensionMethods.EmitSignal[T] (Godot.Object node, System.Object[] args) [0x00000] in /home/maciej/station/Station/src/ExtensionMethods.cs:27 
  at PlayerController._Process (System.Single delta) [0x00069] in /home/maciej/station/Station/src/actors/player/PlayerController.cs:41 
```

**Comment by GamesByKylie** [comment]
I'm having the exact same problem! I'm not calling anything manually, I have "Start Automatically" turned on. As soon as the game starts, it throws this error and stops. By using Debug.Log, I know that the last thing it does is call Dialogue.SetNode() before something goes wrong. I was working with it a few days ago and had no issues. Between then and now, none of the Yarn-related code or objects have been edited. It was working and then suddenly it stopped. I tried several different Yarn files, several of which haven't been changed in weeks (when they worked perfectly) and got the same error for each one.

```
NullReferenceException: Object reference not set to an instance of an object
Yarn.VirtualMachine.SetNode (System.String nodeName) (at <60a9cde088b1430eb5c95ed66926df4f>:0)
Yarn.Dialogue.SetNode (System.String startNode) (at <60a9cde088b1430eb5c95ed66926df4f>:0)
Yarn.Unity.DialogueRunner.<StartDialogue>g__RunDialogue|21_0 (Yarn.Unity.DialogueRunner+<>c__DisplayClass21_0& ) (at Assets/YarnSpinner/Scripts/DialogueRunner.cs:256)
Yarn.Unity.DialogueRunner.StartDialogue (System.String startNode) (at Assets/YarnSpinner/Scripts/DialogueRunner.cs:240)
Yarn.Unity.DialogueRunner.StartDialogue () (at Assets/YarnSpinner/Scripts/DialogueRunner.cs:225)
Yarn.Unity.DialogueRunner.Start () (at Assets/YarnSpinner/Scripts/DialogueRunner.cs:537)
```

**Comment by mgilski** [comment]
I explained the solution to my problem in the attached pull request - make sure that both `Dialogue.LogDebugMessage` and `Dialogue.LogErrorMessage` have something assigned to them.

**Comment by desplesda** [comment]
Resolved in afc2ff426787917e0c3cb9ab8310c7957a3a9730. Thank you!
